x/ccv/provider/keeper/relay.go

```diff
@@ -213,57 +216,68 @@ func (k Keeper) EndBlockCIS(ctx sdk.Context) {
 	k.SetValsetUpdateBlockHeight(ctx, valUpdateID, uint64(ctx.BlockHeight()+1))
 }
 
-// OnRecvSlashPacket slashes and jails the given validator in the packet data
+// OnRecvSlashPacket receives a slash packet, validates it, then handles it if valid.
```

Comment:
nit: the receive is done in the IBC module (hence the name, `OnRecv...`)
```suggestion
// OnRecvSlashPacket delivers a received slash packet: validates it and then handles it if valid.
```